plugins/flytekit-pydantic/tests/test_type_transformer.py

```diff
@@ -0,0 +1,203 @@
+    ],
+)
+def test_transform_round_trip(python_type: Type, kwargs: Dict[str, Any]):
+    """Test that a (de-)serialization roundtrip results in the identical BaseModel."""
+    from flytekit.core.context_manager import FlyteContextManager
+
+    ctx = FlyteContextManager().current_context()
+
+    type_transformer = BaseModelTransformer()
+
+    python_value = python_type(**kwargs)
+
+    literal_value = type_transformer.to_literal(
+        ctx,
+        python_value,
+        python_type,
+        type_transformer.get_literal_type(python_value),
+    )
+
+    reconstructed_value = type_transformer.to_python_value(ctx, literal_value, type(python_value))
+
+    # assert reconstructed_value == python_value
+
+
+@pytest.mark.parametrize(
+    "config_type,kwargs",
+    [
+        (Config, {"model_config": {"foo": TrainConfig(loss="mse")}}),
+        (ConfigRequired, {"model_config": {"foo": TrainConfig(loss="mse")}}),
+        (ConfigWithFlyteFiles, {"flytefiles": ["s3://foo/bar"]}),
+        (ConfigWithFlyteDirs, {"flytedirs": ["s3://foo/bar"]}),
+        (ConfigWithPandasDataFrame, {"df": pd.DataFrame({"a": [1, 2, 3], "b": [4, 5, 6]})}),
+    ],
+)
+def test_pass_to_workflow(config_type: Type, kwargs: Dict[str, Any]):
+    """Test passing a BaseModel instance to a workflow works."""
+    cfg = config_type(**kwargs)
+
+    @flytekit.task
+    def train(cfg: config_type) -> config_type:
+        return cfg
+
+    @flytekit.workflow
+    def wf(cfg: config_type) -> config_type:
+        return train(cfg=cfg)
+
+    returned_cfg = wf(cfg=cfg)
+
+    # assert returned_cfg == cfg
+    # TODO these assertions are not valid for all types
+
+
+@pytest.mark.parametrize(
+    "kwargs",
+    [
+        {"flytefiles": ["tests/folder/test_file1.txt", "tests/folder/test_file2.txt"]},
+    ],
+)
+def test_flytefiles_in_wf(kwargs: Dict[str, Any]):
+    """Test passing a BaseModel instance to a workflow works."""
+    cfg = ConfigWithFlyteFiles(**kwargs)
+
+    @flytekit.task
+    def read(cfg: ConfigWithFlyteFiles) -> str:
+        with open(cfg.flytefiles[0], "r") as f:
+            return f.read()
+
+    @flytekit.workflow
+    def wf(cfg: ConfigWithFlyteFiles) -> str:
+        return read(cfg=cfg)  # type: ignore
+
+    string = wf(cfg=cfg)
+    assert string in {"foo", "bar"}  # type: ignore
+
+
+@pytest.mark.parametrize(
+    "kwargs",
+    [
+        {"flytedirs": ["tests/folder/"]},
+    ],
+)
+def test_flytedirs_in_wf(kwargs: Dict[str, Any]):
+    """Test passing a BaseModel instance to a workflow works."""
+    cfg = ConfigWithFlyteDirs(**kwargs)
+
+    @flytekit.task
+    def listdir(cfg: ConfigWithFlyteDirs) -> List[str]:
+        return os.listdir(cfg.flytedirs[0])
+
+    @flytekit.workflow
+    def wf(cfg: ConfigWithFlyteDirs) -> List[str]:
+        return listdir(cfg=cfg)  # type: ignore
+
+    dirs = wf(cfg=cfg)
+    assert len(dirs) == 2  # type: ignore
+
+
+def test_double_config_in_wf():
```

Comment:
I used this to make sure that the flyte object store isnt for whatever reason mixed up when deserializing two otherwise similar objects. Think it could catch something in the future with a test that aims to ensure accuracy when multiple basermodels are passed in the wf kwargs.